### PR TITLE
fix float printout

### PR DESCRIPTION
**Pull request by mdehoon**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This PR fixes a doctest failure in `Bio.pairwise2` that shows up when running `python run_tests.py doctest`.

### CODE REVIEW

File: Tests/test_pairwise2_no_C.py [hunk 1]
```diff
@@ -16,11 +16,18 @@
 
 from Bio import pairwise2
 
-# Explicitly using pure Python fallback functions:
-pairwise2._make_score_matrix_fast = pairwise2._python_make_score_matrix_fast
-pairwise2.rint = pairwise2._python_rint
-
```
Comment (peterjc) [inline Tests/test_pairwise2_no_C.py:22]: Did this have any functional usage? i.e. Might anyone be assuming these exist?

Comment (mdehoon) [reply Tests/test_pairwise2_no_C.py:22]: No, this script does not get imported anywhere.

Comment (peterjc) [reply Tests/test_pairwise2_no_C.py:22]: Sorry, missed this was in the tests folder.

**Comment by mdehoon** [comment]
Just to explain why this PR was needed:

`Bio.pairwise2` has an implementation in C and an implementation in Python.
The C-version is using floating-point values for scores, the Python version is using integers.
The doctest in `Bio.pairwise2` assumed integers.
Running the doctest as part of `python run_tests.py --offline` runs `test_pairwise2_no_C.py` before running the doctest. In `test_pairwise2_no_C.py`, the Python implementation was used; this was not reset at the end of the test. Then in the `Bio.pairwise2` doctest, the Python implementation is still being used; the doctest does not raise an error as it expects integers.
However, running the doctest by itself as `python run_tests.py doctest` uses the C implementation, which using floating-point values. The doctest then fails.

This PR does three things:
- Ensure that the implementation to be used is reset at the end of `test_pairwise2_no_C.py`;
- Change the Python implementation to use floating-point values to be consistent with the C implementation;
- Change the doctest to expect floating-point values.

**Comment by peterjc** [comment]
That's much clearer - would have been perfect for the merge commit comment in hindsight.

**Pull request merged**